app/src/main/kotlin/com/wire/android/ui/home/conversations/details/GroupConversationDetailsViewModel.kt

```diff
@@ -162,9 +168,29 @@ class GroupConversationDetailsViewModel @Inject constructor(
                 }
             }
         }
-        requestInProgress = false
     }
 
+    fun clearConversationContent(dialogState: DialogState) {
+        with(dialogState) {
+            executeWithProgress {
+                val result = withContext(dispatcher.io()) { clearConversationContentUseCase(conversationId) }
+                clearContentSnackbarResult(result, conversationTypeDetail)
+            }
+        }
+    }
+
+    private suspend fun clearContentSnackbarResult(
+        clearContentResult: ClearConversationContentUseCase.Result,
+        conversationTypeDetail: ConversationTypeDetail
+    ) {
+        if (conversationTypeDetail is ConversationTypeDetail.Connection) throw IllegalStateException("Unsupported conversation type to clear content, something went wrong?")
```

Comment:
do we really want to throw anything here? Does it crash the app? Maybe just "Something went wrong error message" and logs would be better?

Reply:
in my opinion the issue is with Connection being ConversationTypeDetail, we almost never treat is a a conversation but as some seperate object family, I think forcing a crash makes sense as if we invoke this on a Connection we are doing something wrong and it could save us from implementing something weird into the production, proper solution would be to exlucde this Connection from ConversationTypeDetail i think.